### ISSUE TITLE
How to set the output directory?

### ISSUE THREAD

**Issue by phiweger** [spacegraphcats/spacegraphcats#223]
I.e. can it be set in the `config.yaml` and if so, what key is needed?

**Comment by phiweger** [comment]
After some inquiry, I think it probably is set by e.g. "catlas_base: dory" in the config.yaml -- So this starts the workflow:

```bash
python -m spacegraphcats config.yaml search
```

This is problematic, if you pass something like "catlas_base: /tmp/dory" because the following rule in the Snakefile will break:

```python
rule bcalm_cdbg:
     input:
        "bcalm.{catlas_base}.k{ksize}.inputlist.txt"
     output:
        "{catlas_base}/bcalm.{catlas_base}.k{ksize}.unitigs.fa"
     shell:
        "(bcalm -in bcalm.{catlas_base}.k{ksize}.inputlist.txt -out {catlas_base}/bcalm.{catlas_base}.k{wildcards.ksize} -kmer-size {wildcards.ksize} -abundance-min 1 >& {output}.log.txt && rm -f {catlas_base}/bcalm.{catlas_base}.k{wildcards.ksize}.{{h5,unitigs.fa.glue*}}) || printf '**\\n**\\n** Cannot run BCALM 2! Please install it and make sure it is on your path!\\n**\\n**\\n'"
```

I think this might be because of some filename issue? I.e. the case above it would try to create `bcalm./tmp/dory.k21_r1` -- which snakemake does not seem to like. I get an output directory with a bcalm log, which contains following message:

```bash
less /tmp/dory/bcalm./tmp/dory.k21.unitigs.fa.log.txt

BCALM 2, git commit c8ac60252fa0b2abf511f7363cff7c4342dac2ee
setting storage type to hdf5
HDF5-DIAG: Error detected in HDF5 (1.8.18) thread 0:
  #000: /opt/conda/conda-bld/bcalm_1554291614190/work/gatb-core/gatb-core/thirdparty/hdf5/src/H5F.c line 604 in H5Fopen(): unable to open file
    major: File accessibilty
    minor: Unable to open file
...
```

Long story short: Is there a way to just pass an `--outdir` either to the command or the config? And treat `catlas_base` as a simple `--prefix`? Ie not mix the two?

Thanks a lot for your help,
Adrian

**Comment by phiweger** [comment]
My use case is to run spacegraphcats from a docker container, maybe this explains more of why I have opened the issue :)

**Comment by taylorreiter** [comment]
+1 -- I've wrapped spacegraphcats in another snakefile, and I would love to be able to control the output location from the command line or from the config file

**Comment by taylorreiter** [comment]
+1 again -- I want to run spacegraphcats in on an hpc that is faster when I write to a scratch dir. @ctb do you have a mental model of how this could be implemented? I'm happy to do it but I'm not sure how/where to start

**Comment by ctb** [comment]
No clear idea. Should be relatively straightforward, but I don't know where to begin :). Will think on't.

**Comment by ctb** [comment]
Try out #251, and let me know!
